tests/test_modeling_ctrl.py

```diff
@@ -172,6 +178,39 @@ def create_and_check_lm_head_model(self, config, input_ids, input_mask, head_mas
                 list(result["lm_logits"].size()), [self.batch_size, self.seq_length, self.vocab_size]
             )
 
+        def create_and_check_generate_lm_head_model(self, config, input_ids, *args):
+            model = CTRLLMHeadModel(config)
+            model.to(torch_device)
+            model.eval()
+
+            # generate function should not produce any None values so that every output is decodable
+            with self.parent.assertRaises(AssertionError):
+                self.check_tokens(
+                    model.generate(max_length=self.max_length)
+                )  # no input, should throw error because bos_token_id is not defined in ctrl
+            self.check_tokens(
+                model.generate(input_ids, num_return_sequences=self.num_return_sequences)
+            )  # batch_size > 1, input_ids defined
+            self.check_tokens(
+                model.generate(input_ids, num_return_sequences=self.num_return_sequences, do_sample=False)
+            )  # batch_size > 1, greedy decoding, input_ids defined
+            self.check_tokens(
+                model.generate(input_ids, num_beams=self.num_beams, num_return_sequences=self.num_return_sequences)
+            )  # num_beams > 1
+            self.check_tokens(
+                model.generate(
+                    input_ids,
+                    do_sample=False,
+                    num_beams=self.num_beams,
+                    num_return_sequences=self.num_return_sequences,
+                )
+            )  # greedy decoding
+
+        def check_tokens(self, output_ids):
+            for token_id in output_ids[0].tolist():
+                self.parent.assertGreaterEqual(token_id, 0)
+                self.parent.assertLess(token_id, self.vocab_size)
+
```

Comment:
I guess this doesn't have to be here anymore?

Reply:
yes, will delete all of those!